### ISSUE TITLE
Cool Project

### ISSUE THREAD

**Issue by ederwander** [alcapriles/Music-Box#2]
Just one idea, why not to do all in real-time  ?

**Comment by alcapriles** [comment]
This is great. I feel like doing it. It's a much cooler idea than the buttons we were using.
Any tips on how to implement it?

Obs: Another idea I have in mind is to change the poor algorithm we have now to a Machine Learning one that better identifies the musical notes on top of the frequencies brought by the chromagram.

**Comment by ederwander** [comment]
Yeah chromagram sucks to extract frequencies, pitch track is still a not solved problem, you understand how one note can be mathematically defined ?  if you understood you will see that chromagram is not be the choice for accurate results.

I can sugest you first get experience in monophonic sounds, polyphonic is a very hard way, course it is possible if you know how do some tricks in frequency domain. The basic idea is if you can get the right pitch frequency you can convert to a note name, maybe just a nice pitch track can kick the ass of machine learning ...

I write a lot of codes to extract pitch in time domain just for monophonic sounds, and its can run in real-time lol.

For polyphonic pitch extraction I not did many things, I never needed!

Just curious in my quick look in some chunks of your code i can see words in portuguese (Portugal, Angola, Brazil?)

Here is not be the best place to discuss and show you some tips, if do you need you can send me email: ederwander@gmail.com
